src/main/scala/generatortools/io/Utils.scala

```diff
@@ -0,0 +1,156 @@
+// See LICENSE for license details.
+
+package generatortools.io
+
+import chisel3._
+import chisel3.experimental._
+import chisel3.internal.firrtl._
+import chisel3.util.Cat
+import dsptools.numbers._
+import firrtl.constraint.IsKnown
+
+//scalastyle:off cyclomatic.complexity method.length
+object ConvertType {
+  /** Convert special type (i.e. Clock, Interval) to equivalent base Chisel type.
+    * Otherwise, keep the current base Chisel type. Warning: Any kind of Record becomes a CustomBundle!
+    */
+  def apply[T <: Data](tpe: T): Data = {
+    val newTpe = tpe match {
+      // TODO: special case others
+      case s: SInt =>
+        s.widthOption match {
+          case None => SInt(64.W)
```

Comment:
These rules all seem a bit scary and arbitrary. Is this actually used/necessary? It seems like a `cloneType` that wants to enforce having a width for some types, but not others. Maybe these places should be replaced with `???`

Reply:
I'm going to take another crack at this this evening. I may pull this all out and up into fft-generator until it all gets tested and worked out

Reply:
I'm going to move the whole generatortools package up into fft-generator until it settles down